### ISSUE TITLE
Spike trap spawned ontop lantern

### ISSUE THREAD

**Issue by MicroChasm** [nsmryan/RustRoguelike#433]
I died in a game because I stepped on a torch tile that also had a spike trap underneath it. While the spike trap is uniquely difficult to see under a torch, I think it would be best to just not spawn traps under torches. 

[action_log.txt](https://github.com/nsmryan/RustRoguelike/files/8238586/action_log.txt)

**Comment by MicroChasm** [comment]
I also had a situation where a spike trap spawned on the same tile as a freeze trap, and both triggered when I stepped on the tile.

**Comment by nsmryan** [comment]
I think I see where this problem occurs- when checking for whether to use a tile, I only checked whether there was an entity that blocks movement, but not for just any entity.
Now traps, triggers, and other things that are placed on the map will only use completely empty tiles, not tiles that have traps, triggers or items.

I also moved grass generation to below all these other things, as grass is now an entity (in order to animate it) and this would prevent, for example, golems or items spawning on grass. Moving it lower in generation allows grass to place on top of other entities.
